test/server/cards/02_SHD/Headhunting.spec.js

```diff
@@ -0,0 +1,132 @@
+                });
+            });
+
+            it('should attack with three units', function () {
+                this.player1.clickCard(this.headhunting);
+
+                // first attack, bounty hunter
+                expect(this.player1).toBeAbleToSelectExactly([this.atst, this.reputableHunter, this.theMandalorian, this.wampa]);
+                expect(this.player1).toHaveEnabledPromptButton('Pass ability');
+                this.player1.clickCard(this.reputableHunter);
+                expect(this.player1).toBeAbleToSelectExactly([this.bountyGuildInitiate, this.consularSecurityForce]);
+                expect(this.player1).toHaveEnabledPromptButton('Pass ability');
+                this.player1.clickCard(this.consularSecurityForce);
+                expect(this.reputableHunter.exhausted).toBe(true);
+                expect(this.consularSecurityForce.damage).toBe(5);
+                expect(this.reputableHunter.damage).toBe(3);
+
+                // second attack, non-bounty-hunter
+                this.consularSecurityForce.damage = 0;
+                expect(this.player1).toBeAbleToSelectExactly([this.atst, this.theMandalorian, this.wampa]);
+                expect(this.player1).toHaveEnabledPromptButton('Pass ability');
+                this.player1.clickCard(this.atst);
+                expect(this.player1).toBeAbleToSelectExactly([this.bountyGuildInitiate, this.consularSecurityForce]);
+                expect(this.player1).toHaveEnabledPromptButton('Pass ability');
+                this.player1.clickCard(this.consularSecurityForce);
+                expect(this.atst.exhausted).toBe(true);
+                expect(this.consularSecurityForce.damage).toBe(6);
+                expect(this.atst.damage).toBe(3);
+
+                // third attack, leader bounty hunter
+                this.consularSecurityForce.damage = 0;
+                expect(this.player1).toBeAbleToSelectExactly([this.theMandalorian, this.wampa]);
+                expect(this.player1).toHaveEnabledPromptButton('Pass ability');
+                this.player1.clickCard(this.theMandalorian);
+                expect(this.player1).toBeAbleToSelectExactly([this.bountyGuildInitiate, this.consularSecurityForce]);
+                expect(this.player1).toHaveEnabledPromptButton('Pass ability');
+                this.player1.clickCard(this.consularSecurityForce);
+                expect(this.theMandalorian.exhausted).toBe(true);
+                expect(this.consularSecurityForce.damage).toBe(6);
+                expect(this.theMandalorian.damage).toBe(3);
+
+                expect(this.player2).toBeActivePlayer();
+            });
+
+            it('should be able to be passed', function () {
+                this.player1.clickCard(this.headhunting);
+
+                // first attack - skip target resolution, go straight to next attack
+                expect(this.player1).toBeAbleToSelectExactly([this.atst, this.reputableHunter, this.theMandalorian, this.wampa]);
```

Comment:
I think we need a way to differentiate the passes here. If someone plays this card and then passes, why is it passing the first attack and not the whole effect? This can be a TODO, but I would imagine we might need options like "Pass attack" vs. "Pass all attacks" if we don't want to make them choose the amount of attacks up front.

This brings up an interesting case Im not sure of. I know there are windows for trigger resolution within an attack, but if a unit is defeated by the first attack of Headhunting, does its When Defeated resolve before the second attack? I assume all of them are resolved at the end of the whole event resolution.

Reply:
Will discuss the first question in the UI chat. For the second question, the on defeat will happen in the next available trigger window (so immediately after the unit is defeated). Per this rule:

![IMG_6590](https://github.com/user-attachments/assets/b491683c-6cac-4813-9650-492673339d5d)

Reply:
Though right now the implementation is buggy, but I'm planning a future PR focused on windows and timing so I'll address it then.

Reply:
Actually maybe I will just put a TODO for the first one and then trigger the discussion, since this is blocking some stuff.